src/main/java/org/duckdb/DuckDBDatabaseMetaData.java

```diff
@@ -171,27 +171,76 @@ public String getIdentifierQuoteString() throws SQLException {
 
     @Override
     public String getSQLKeywords() throws SQLException {
-        throw new SQLFeatureNotSupportedException("getSQLKeywords");
+        Statement statement = conn.createStatement();
+        statement.closeOnCompletion();
+        ResultSet rs = statement.executeQuery(
+                "SELECT keyword_name FROM duckdb_keywords()");
+        StringBuilder sb = new StringBuilder();
+        while (rs.next()) {
+            sb.append(rs.getString(1));
+            sb.append(',');
+        }
+        return sb.toString();
     }
 
     @Override
     public String getNumericFunctions() throws SQLException {
-        throw new SQLFeatureNotSupportedException("getNumericFunctions");
+        Statement statement = conn.createStatement();
+        statement.closeOnCompletion();
+        ResultSet rs = statement.executeQuery(
+                "SELECT DISTINCT function_name FROM duckdb_functions() " +
+                        "WHERE parameter_types[1] ='DECIMAL'" +
+                        "OR parameter_types[1] ='DOUBLE'" +
+                        "OR parameter_types[1] ='SMALLINT'" +
+                        "OR parameter_types[1] = 'BIGINT'");
+        StringBuilder sb = new StringBuilder();
+        while (rs.next()) {
+            sb.append(rs.getString(1));
+            sb.append(',');
+        }
+        return sb.toString();
     }
 
     @Override
     public String getStringFunctions() throws SQLException {
-        throw new SQLFeatureNotSupportedException("getStringFunctions");
+        Statement statement = conn.createStatement();
+        statement.closeOnCompletion();
+        ResultSet rs = statement.executeQuery(
+                "SELECT DISTINCT function_name FROM duckdb_functions() WHERE parameter_types[1] = 'VARCHAR'");
+        StringBuilder sb = new StringBuilder();
+        while (rs.next()) {
+            sb.append(rs.getString(1));
+            sb.append(',');
+        }
+        return sb.toString();
     }
 
     @Override
     public String getSystemFunctions() throws SQLException {
-        throw new SQLFeatureNotSupportedException("getSystemFunctions");
+        Statement statement = conn.createStatement();
+        statement.closeOnCompletion();
+        ResultSet rs = statement.executeQuery(
+                "SELECT DISTINCT function_name FROM duckdb_functions() WHERE parameter_types[1] IS NULL");
```

Comment:
This condition doesn't quite work - probably need to exclude vararg functions somehow?

Reply:
```sql 
select DISTINCT function_name from duckdb_functions() where length(parameter_types) = 0
```
??

Reply:
It's not used consistently??

```sql 
D select DISTINCT function_name, parameter_types, varargs from duckdb_functions() where parameter_types[1] is null;
```

|           function_name            |  parameter_types   | varargs |
|------------------------------------|--------------------|---------|
| map_from_entries                   | []                 | ANY     |
| list_median                        | [NULL]             |         |
| ... | ... | ... |

Sometimes, they have a type in the Vargas columns; other times, they have an explicit `NULL` as the first argument.

Reply:
@Mause is this ok solution given the above, given that vargargs aren't handled consistently?

Reply:
Sounds good for now yeah